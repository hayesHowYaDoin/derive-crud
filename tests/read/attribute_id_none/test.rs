use derive_crud::Read;

#[derive(Read)]
#[crud_table("test_table")]
struct Test {
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    let database_url = std::env::var("DATABASE_URL").unwrap();
    let pool = sqlx::SqlitePool::connect(&database_url).await.unwrap();

    let _ = Test::read(&pool, 1);
}
