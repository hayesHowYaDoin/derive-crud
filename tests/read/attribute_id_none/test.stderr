error: A struct field must be annotated with `#[crud_id]` to specify the database table
 --> tests/read/attribute_id_none/test.rs:5:13
  |
5 |   struct Test {
  |  _____________^
6 | |     id: i64,
7 | |     name: String,
8 | | }
  | |_^

error[E0599]: no function or associated item named `read` found for struct `Test` in the current scope
  --> tests/read/attribute_id_none/test.rs:15:19
   |
5  | struct Test {
   | ----------- function or associated item `read` not found for this struct
...
15 |     let _ = Test::read(&pool, 1);
   |                   ^^^^ function or associated item not found in `Test`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `read`, perhaps you need to implement one of them:
           candidate #1: `derive_crud::futures_util::AsyncReadExt`
           candidate #2: `digest::XofReader`
           candidate #3: `sqlx_core::net::socket::Socket`
           candidate #4: `std::io::Read`
           candidate #5: `tokio::io::AsyncReadExt`

error[E0599]: no function or associated item named `read_one` found for struct `Test` in the current scope
  --> tests/read/attribute_id_none/test.rs:16:19
   |
5  | struct Test {
   | ----------- function or associated item `read_one` not found for this struct
...
16 |     let _ = Test::read_one(&pool, 1);
   |                   ^^^^^^^^ function or associated item not found in `Test`

error[E0599]: no function or associated item named `read_all` found for struct `Test` in the current scope
  --> tests/read/attribute_id_none/test.rs:17:19
   |
5  | struct Test {
   | ----------- function or associated item `read_all` not found for this struct
...
17 |     let _ = Test::read_all(&pool);
   |                   ^^^^^^^^ function or associated item not found in `Test`
